src/biocommons/seqrepo/fastadir/fastadir.py

```diff
@@ -50,13 +51,18 @@ class FastaDir(BaseReader, BaseWriter):
 
     """
 
-    def __init__(self, root_dir, writeable=False, check_same_thread=True, fd_cache_size=0):
+    def __init__(
+        self,
+        root_dir: str,
+        writeable: bool = False,
+        check_same_thread: bool = True,
+        fd_cache_size: Optional[int] = 0,
+    ) -> None:
         """Creates a new sequence repository if necessary, and then opens it"""
 
         self._root_dir = root_dir
         self._db_path = os.path.join(self._root_dir, "db.sqlite3")
         self._writing = None
-        self._db = None
```

Comment:
Same comment as other PR, should this be separated out in a different issue?

Reply:
I sort of intended this PR to *be* that issue. This is a line that doesn't do anything except establish a type annotation that is problematic for how the entire rest of the class is written. I originally addressed it in the other issue only because it makes pyright explode if you touch anything else in this module, but the only reason it needs to change at all is for type checking reasons, not because it affects function (since the variable is reassigned to a real value four lines down).

Reply:
Ah okay. I didn't pull anything up in my editor so couldn't see any warnings and it just appeared that you were deleting unused code since you were already in that module